parsing_1.py

```diff
@@ -39,4 +36,24 @@ def write_csv(counter):
         file_contents = r.content.decode().replace("\r\n", "\n")
         log.write(file_contents)
 
-write_csv(Counter(parse(file_contents)))
+# Extract the dates from every valid log entry
+date_strs = re.findall(APACHE_LOG_REGEX, log, re.MULTILINE)
+dates = [datetime.strptime(date, APACHE_LOG_DATE_FORMAT) for date in dates]
```

Comment:
This applies `datetime.strptime` to every date in `date_strs` and creates a new list with the results.